lib/manageiq/appliance_console/prompts.rb

```diff
@@ -71,6 +72,11 @@ def ask_for_hostname(prompt, default = nil, validate = HOSTNAME_REGEXP, error_te
       just_ask(prompt, default, validate, error_text, &block)
     end
 
+    def ask_for_messaging_hostname(prompt, default = nil, error_text = "a valid Messaging Hostname (not an IP or localhost)", &block)
+      validation = ->(h) { h =~ MESSAGING_HOSTNAME_REGEXP && h !~ IP_REGEXP }
```

Comment:
Hm I wonder if instead of mostly duplicating the hostname regex we just say `h =~ HOSTNAME_REGEXP && h !~ IP_REGEXP && h != "localhost"` here?

Reply:
not a bad idea, the only difference I see is that the regex detects instances of `localhost` anywhere in the string. i.e. a hostname that has `.localhost` at the end (or anywhere), another example is the infamous `localhost.localadmin`